FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs

```diff
@@ -188,9 +227,21 @@ public async Task CompleteJobAsync(Job job, CancellationToken cancellationToken
                     };
                 }
 
-                await SaveSchedulerMetadata(schedulerSetting, cancellationToken);
+                // Remove the job information that have been resumed and executed.
+                // If the current job fails, save it to unfinishedJob list.
+                var unfinishedJobs = schedulerSetting.UnfinishedJobs.ToList();
+                unfinishedJobs.RemoveAll(unfinished => unfinished.Id == job.ResumedJobId);
```

Comment:
Just want to double confirm if I understand it correctly: even if the job is resumed, it does not mean it is finished, right? If the job still fails, it will be added to `unfinishedJobs` again in Line 237.

Reply:
Yes. For a job that fails multiple times, it will be resume in every following trigger and create a new instance. The difference is that the error message can be recorded as the errors can be different for different trigger. e.g. the first time it fails with auth, and next it fails due a transient error.